### PR TITLE
Create SSM Data Sync resource

### PR DESCRIPTION
**Pull request by gswai123**
This PR creates a Resource Data Sync (AWS Systems Manager) resource for Cloud Custodian, along with a KMS Key filter and a Delete action. A value filter can also be used with the Data Sync resource to in order to ensure that Destination S3 buckets are owned by whitelisted accounts. 

Issue Link: https://github.com/cloud-custodian/cloud-custodian/issues/6621

### CODE REVIEW

File: c7n/resources/ssm.py [hunk 1]
```diff
@@ -597,3 +598,68 @@ def register_resource(cls, registry, resource_class):
...
+        id = 'DataSync'
+        name = 'Title'
+
+    permissions = ('ssm:ListResourceDataSync',)
+
+
+@SSMDataSync.filter_registry.register('kms-key')
+class KmsFilter(KmsRelatedFilter):
+    """
+    Filter a resource by its associcated kms key and optionally the aliasname
```
Comment (kapilt) [inline c7n/resources/ssm.py:622]: can we skip this docstring, we have another pr which is nuking the extant so the baseclass docs can be utilized/updated in a single place.

File: c7n/resources/ssm.py [hunk 2]
```diff
@@ -597,3 +598,68 @@ def register_resource(cls, registry, resource_class):
...
+    schema = type_schema(
+        'delete',
+        force={'type': 'boolean'}
+    )
+    permissions = ('ssm:DeleteResourceDataSync',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('ssm')
+        for r in resources:
+            client.delete_resource_data_sync(SyncName=r['SyncName'])
```
Comment (kapilt) [inline c7n/resources/ssm.py:665]: we should catch resource doesn't exist in this context.

File: c7n/resources/ssm.py (hunk 1 above)
Comment (gswai123) [reply c7n/resources/ssm.py:622]: Got it.

File: c7n/resources/ssm.py [hunk 3]
```diff
@@ -597,3 +598,54 @@ def register_resource(cls, registry, resource_class):
...
+    )
+    permissions = ('ssm:DeleteResourceDataSync',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('ssm')
+        for r in resources:
+            try:
+                client.delete_resource_data_sync(SyncName=r['SyncName'])
+            except client.exceptions.ResourceDataSyncNotFoundException as e:
+                raise(e)
```
Comment (kapilt) [inline c7n/resources/ssm.py:651]: actually we want to continue in this case, we're trying to delete something thats already gone, we can safely continue to the next.

**Comment by kapilt** [comment]
addressed the rebase against the rather large set of conflicts due to ssm document support.

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**